### PR TITLE
Use pylint-django

### PR DESCRIPTION
**Pull request by nedbat**
This reduces a lot of false negatives, especially for Django 1.8

### CODE REVIEW

**Comment by nedbat** [comment]
@singingwolfboy @jcdyer have a look-see.

File: setup.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
     },
 
     install_requires=[
-        'pylint>=1.4.0',
+        'pylint==1.4.4',
+        'pylint-django==0.6.1',
```
Comment (singingwolfboy) [inline setup.py]: Shouldn't these changes be reflected in `requirements.txt`? Frankly, I'm not sure why this project has both a `requirements.txt` file *and* an `install_requires` section in the setup.py (that doesn't read the `requirements.txt` file).

File: setup.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 setup(
     name='edx-lint',
-    version='0.3.0',
+    version='0.3.2',
```
Comment (singingwolfboy) [inline setup.py]: Why the bump from `0.3.0` to `0.3.2`, rather than `0.3.1`? Not a problem, just curious for the rationale.

File: setup.py (hunk 1 above)
Comment (jcdyer) [inline setup.py]: Agreed that requirements and install_requires should be kept in sync.

Comment (nedbat) [inline setup.py]: I've changed requirements.txt to not mention pylint, instead to install setup.py.

File: setup.py (hunk 2 above)
Comment (nedbat) [inline setup.py]: That was an indication that I needed to rebase: master was already at 0.3.1

File: setup.py (hunk 1 above)
Comment (singingwolfboy) [inline setup.py]: In that case, what's the purpose of the `requirements.txt` file? Can that file just be deleted?

Comment (nedbat) [inline setup.py]: requirements.txt is for developers working on edx-lint.  Setup.py is for people using edx-lint.  I thought this was a well-established pattern?  I guess not.

Comment (nedbat) [inline setup.py]: Maybe if I rename it to dev-requirements.txt, we'll be OK?

Comment (singingwolfboy) [inline setup.py]: The pattern that I thought was the most well-established was:
* `requirements.txt` lists dependencies to use the software
* `setup.py` reads `requirements.txt` and passes the contents to the `install_requires` block
* If the software has additional dependencies for development, such as for running tests or building documentation, those dependencies are either listed in the `tests_require` section of `setup.py`, or listed in a secondary requirements file (such as `dev-requirements.txt`) which is similarly read into `setup.py`

Is there some documentation that we can refer to, to determine the correct pattern? I'm willing to be proven wrong.

**Comment by jcdyer** [comment]
:+1:

File: setup.py (hunk 1 above)
Comment (nedbat) [inline setup.py]: There is no correct pattern, there are evolving best practices.  I don't see the advantage of setup.py reading a text file.  I'll rename the file to dev-requirements.txt.  The README will direct them to install from that if they are developing in the repo.  The setup.py will get the correct requirements for installed use.  OK?

Comment (singingwolfboy) [inline setup.py]: That sounds good to me.

Comment (nedbat) [inline setup.py]: 🎼 It's done, gimme some thumb 🎶

**Comment by singingwolfboy** [comment]
:+1:

**Pull request merged**